### PR TITLE
initial draft

### PR DESCRIPTION
**Pull request by snodnipper**
initial draft code for experimentation / collaboration around the topic of map data updates.

Currently tied to OS Unplugged but clearly such behaviour is applicable to other apps.

### CODE REVIEW

File: updater-android/src/main/java/uk/os/unplugged/updater/android/Provider.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+package uk.os.unplugged.updater.android;
+
+import java.io.File;
+
+/**
+ * Note: all references are assumed to have corresponding MD5 files.  For example, foo.bar is
+ * expected to have a corresponding foo.bar.md5 file.
+ */
+public interface Provider {
```
Comment (dhardiman) [inline updater-android/src/main/java/uk/os/unplugged/updater/android/Provider.java]: Do we need a more expressive name for this? Not sure of Java code style, and perhaps the `package` namespace makes it clear, but if not, perhaps this is lacking some context.

File: updater-android/src/main/java/uk/os/unplugged/updater/android/Util.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    private static final String TAG = Util.class.getSimpleName();
+
+    /**
+     * source: http://stackoverflow.com/questions/106770/standard-concise-way-to-copy-a-file-in-java
+     * @param src file
+     * @param dst file
+     * @throws IOException
+     */
+    public static void copy(File src, File dst) throws IOException {
+        InputStream in = new FileInputStream(src);
```
Comment (dhardiman) [inline updater-android/src/main/java/uk/os/unplugged/updater/android/Util.java:47]: Wow, no APIs for this? Guess those patents really suck 😉

**Comment by dhardiman** [comment]
Minor question on the naming of the interface, other than that, LGTM.

File: updater-android/src/main/java/uk/os/unplugged/updater/android/Util.java (hunk 2 above)
Comment (snodnipper) [inline updater-android/src/main/java/uk/os/unplugged/updater/android/Util.java:47]: could always dump the Apache Commons IO truck for this :D

File: updater-android/src/main/java/uk/os/unplugged/updater/android/Provider.java (hunk 1 above)
Comment (snodnipper) [inline updater-android/src/main/java/uk/os/unplugged/updater/android/Provider.java]: how about DataSource or DataProvider?

**Comment by snodnipper** [comment]
Cheers Dave, much appreciated.

**Pull request merged**